src/core/thread/mle.cpp

```diff
@@ -2859,6 +2860,18 @@ otError Mle::HandleChildUpdateRequest(const Message &aMessage, const Ip6::Messag
     // Leader Data, Network Data, Active Timestamp, Pending Timestamp
     SuccessOrExit(error = HandleLeaderData(aMessage, aMessageInfo));
 
+    // Status
+    if (Tlv::GetTlv(aMessage, Tlv::kStatus, sizeof(status), status) == OT_ERROR_NONE)
+    {
+        VerifyOrExit(status.IsValid(), error = OT_ERROR_PARSE);
+
+        if (status.GetStatus() == StatusTlv::kError)
+        {
+            BecomeDetached();
```

Comment:
We should validate that this message actually came from the parent by checking that the IPv6 source address matches the parent's link-local IPv6 address.